scan.py

```diff
@@ -223,21 +266,32 @@ def lambda_handler(event, context):
     create_dir(os.path.dirname(file_path))
     s3_object.download_file(file_path)
 
-    to_download = clamav.update_defs_from_s3(
-        s3_client, AV_DEFINITION_S3_BUCKET, AV_DEFINITION_S3_PREFIX
-    )
+    filename, file_extension = os.path.splitext(file_path)
+    if file_extension == '.pdf':
+        try:
+            xmldoc, infected = PDFiD(file_path, raise_exceptions=False, return_cleaned=True)
+        except:
+            print('Exception occured while scanning PDF by PDFiD')
```

Comment:
It looks like a file can potentially fail scanning by PDFiD but pass virus scan. If we are not able to pass both the scans, the file should not be marked as safe. Currently our tax upload UI is minimalistic and there is no way to indicate a file is unreadable. Maybe we can stop the execution here and send an email if either of the scans fail ?

Reply:
The caveat here is that the PDFiD scan will throw an Exception if the PDF file is not a valid PDF (at least, that has been observed till now). The first check that goes here is the check for the extension, which if not a PDF, to just pass the object to the virus scan. Note that we have other kinds of objects also being upoaded to S3 (under bill preview, for eg.), and this function is invoked for all kinds of objects being uploaded to the bucket. Since we want only the valid PDFs to go through this check for an embedded action scan, the file extension check and the Exception block takes care of this. The virus scan should happen on all the objects, only unless a valid PDF is indicated to have embedded JS/actions. This is what I have tried to solve here. 

cc: @Harishgolla